src/main/java/net/minecraftforge/fml/packs/ModFileResourcePack.java

```diff
@@ -66,7 +67,7 @@ protected InputStream getInputStream(String name) throws IOException
     {
         final Path path = modFile.getLocator().findPath(modFile, name);
         if(!Files.exists(path))
-            throw new ResourcePackFileNotFoundException(path.toFile(), name);
+            throw new FileNotFoundException(String.format("'%s' in ResourcePack '%s'", path, name));
```

Comment:
If I read vanilla code correctly, the path should not point to the zip entry, but to the file of the pack (which is the mod file in this case). So you can do `modFile.getFilePath().toFile()` which should be fine as the mod jar always exists, and use the vanilla exception